### DISCUSSION TITLE
DuplexChannelFactory only 3 ctors are visible out of 21

### DISCUSSION THREAD

**Discussion by 11v1** [CoreWCF/CoreWCF#1251] [Q&A]
I'm moving my project from net4.8 to net7.0. The problem is that DuplexChannelFactory IntelliSense shows only 3 constructors, while for net4.8 it shows 21. In the sources of System.ServiceModel.DuplexChannelFactory I see a lot of public constructors, which are not visible when I add System.ServiceModel.Primitives nuget v6.0.0 to my projects. I need the following ctor to be available:
`public DuplexChannelFactory(Type callbackInstanceType, Binding binding, EndpointAddress remoteAddress)`

What am I doing wrong?

**Comment by mconnew** [comment]
This is an issue for https://github.com/dotnet/wcf  
CoreWCF is only the service side.
